src/realm/sync/network/default_socket.cpp

```diff
@@ -491,70 +492,69 @@ void DefaultSocketProvider::start()
 {
     std::unique_lock<std::mutex> lock(m_mutex);
     // Has the thread already been started or is running
-    if (m_state == State::Starting || m_state == State::Started || m_state == State::Running)
+    if (m_state == State::Starting || m_state == State::Running)
         return; // early return
 
     // If the thread has been previously run, make sure it has been joined first
-    if (m_state == State::Stopping || m_state == State::Stopped) {
```

Comment:
I had these lines in here in case `stop()` (without waiting) and then `start()` was called, but we never do this in the code; and I am working to change the behavior of stop() being called, in general, so this should be fine.